### DISCUSSION TITLE
Where is the SPRING_DATASOURCE_URL of dolphinscheduler_env.sh used?

### DISCUSSION THREAD

**Discussion by chaclus** [apache/dolphinscheduler#12905] [Q&A]
dolphinscheduler version: 3.1.1
linux version: CentOS Linux release 7.9.2009

The configuration information about the database in the dolphinscheduler_env.sh file is declared in the environment variables. Excuse me, where are these variables used? Does the database configuration in the dolphinscheduler_env.sh file override the data in application.yaml in master-server and tools configure? If it is overridden, which part of the source code can see the overriding logic? I have configured the database information in the dolphinscheduler_env.sh file during cluster deployment. The following exception is thrown when executing the "bash tools/bin/upgrade-schema.sh" script. I reconfigured the database information in the tools/conf/application.yaml file, but still reported an error. May I ask if I have overlooked the configuration somewhere?

dolphinscheduler_env.sh 
<img width="1220" alt="image" src="https://user-images.githubusercontent.com/6264879/201906111-c9da370e-6fe9-4e74-b492-36a8eeb632aa.png">

tools/conf/application.yaml
<img width="944" alt="image" src="https://user-images.githubusercontent.com/6264879/201906326-393c93d5-f202-43a4-83b4-1fee5d8ea27b.png">

the exception when execute tools/bin/upgrade-schema.sh script
<img width="1552" alt="image" src="https://user-images.githubusercontent.com/6264879/201906616-28dd7f03-3b00-469f-b507-27cdcd091414.png">

**Comment by DarkAssassinator** [comment]
hi @vainly, this is Spring boot logic, Environment variables will override yml files. May u can check the Spring doc: https://docs.spring.io/spring-boot/docs/1.5.6.RELEASE/reference/html/boot-features-external-config.html

**Reply by chaclus** [reply]
Thank you for your reply, I'll try running it locally.
